var/spack/repos/builtin/packages/patchelf/package.py

```diff
@@ -18,3 +22,23 @@ class Patchelf(AutotoolsPackage):
     version('0.10', sha256='b2deabce05c34ce98558c0efb965f209de592197b2c88e930298d740ead09019')
     version('0.9',  sha256='f2aa40a6148cb3b0ca807a1bf836b081793e55ec9e5540a5356d800132be7e0a')
     version('0.8',  sha256='14af06a2da688d577d64ff8dac065bb8903bbffbe01d30c62df7af9bf4ce72fe')
+
+    def test(self):
+        patchelf = which('patchelf')
+        assert patchelf is not None
+
+        tty.msg('test: Ensuring use of the installed executable')
+        patchelf_dir = os.path.dirname(patchelf.path)
+        assert patchelf_dir == self.prefix.bin
+
+        tty.msg('test: Checking version')
+        output = patchelf('--version', output=str.split, error=str.split)
+        assert output.strip() == 'patchelf {0}'.format(self.spec.version)
```

Comment:
These kind of tests seem to be recurring:
1. Check that we find an executable
2. Check that the executable is in the prefix
3. Check version using a regex

so it could be worth considering to design a more concise way to express that a package needs to perform them on a set of executables.

Reply:
Agreed. I think we've covered most of that, but not the check for whether it's in the prefix, with the `run_test` method. We'll keep working on refactoring common cases.

Reply:
In the newest commit, I believe the `run_test` method covers all cases in this test, and I've refactored this test to use it.